### PR TITLE
Core/Combat: Disable triggers entering combat

### LINKED ISSUE #24290: stuck on combat with icecrown citadel traps (for sure it needs generic fix)
**Description:**

people get stuck on combat if they walk on the traps when enabled and they don't die.

**Current behaviour:**

you end stuck on combat.

**Expected behaviour:**

you don't enter combat with them

**Steps to reproduce the problem:**

1. .go xyz 4228.416504 2768.950195 350.963074 631
2. .cheat god on
3. .gm off
3. walk around enabled traps

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

70569ae3e66e

**Operating system:** debian 10 x64

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Core/Combat: Disable triggers entering combat by adding a new creature_template.flags_extra value 0x00002000 CREATURE_FLAG_EXTRA_NO_COMBAT

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #24290 , #23799

**Tests performed:** (Does it build, tested in-game, etc.)
- Tested #24290
- Tested #23799

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] List the triggers that need to be disallowed to enter combat

### CODE REVIEW

**Comment by Jildor** [comment]
this can break things like halion controller: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp#L608-L638

and blood prince council controller: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_blood_prince_council.cpp#L258-L307

and probably valithria controller: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_valithria_dreamwalker.cpp#L565

I'm not sure how many triggers like this can be, if we can't find a way for get all controller and assing a custom flag, a lot of boss encounters and maybe quests (I'm not sure), can break.

Another safer way would be to make a custom flag that would avoid entering combat, but it would have to be assigned to many triggers.

**Comment by Keader** [comment]
Old Behavior: Triggers can enter combat and remove combat like any other npc.
So 90% of scripts is based on that.
After treeston threat rewrite, triggers cant automatic enter in combat (you need force they enter in combat and you need force they  remove combat).
I belive that both issues here, is caused by some negative spell (casted by trigger) put they in combat and nothing remove (caused by treeston rewrite).
To merge this issue, need check all current scripts that use triggers to control fights.
We have Illidari Council (Black Temple) and Yogg Saron (Ulduar), with this behavior too.
Anyway... if i remember right, has a combat trigger (that keep players in combat with him) in Northrend Beasts (TOC) and Reliquary of Souls/Lost (Black Temple) and they're blizzlike keep players in combat while fighting.

**Comment by jackpoz** [comment]
Do those triggers use spells to put players in combat or scripts ? We could skip entering combat when casting spells for triggers.

**Comment by Ovahlord** [comment]
It varies. There are some trigger npcs that engage players via aggro creator spell but there are also simple trigger npcs that use the   combat pulse mechanic of bosses.

**Comment by jackpoz** [comment]
I would rather have triggers not enter combat by default and allow that with a flag for the ones that need to do that (with the AI taking care of adding/removing the flag as needed)

**Comment by Warpten** [comment]
Summoned triggers probably enter combat as they are summoned since they usually despawn on encounter end

**Comment by jackpoz** [comment]
Implemented a new creature_template.flags_extra value CREATURE_FLAG_EXTRA_NO_COMBAT 0x00002000 that doesn't allow the creature to enter combat.
This can be added to all triggers with a sql query, or only to the triggers we want if we get a list. For now only the triggers in the 2 known issues that this PR addresses are handled (please list more)

**Comment by offl** [comment]
In case anyone wants to add this flag to more creatures, use https://gist.github.com/Kinzcool/e81137dab3bc828631ada717f7da5159
And
```sql
SELECT * FROM creature_template a LEFT JOIN dbc_creaturedifficulty b ON a.entry=b.creatureid WHERE b.flags1 & 0x02000000 /*CREATURE_DIFFICULTYFLAGS_IGNORE_COMBAT*/
```
Note: some of them may be not valid at all or can broke some things

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+--
+UPDATE creature_template SET flags_extra = flags_extra | 0x00002000 /* CREATURE_FLAG_EXTRA_NO_COMBAT */ WHERE entry IN
+(37744 -- ICC Frost Freeze Trap
+,37013 -- ICC Rotface Puddle Stalker
```
Comment (Jildor) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql:4]:
```diff
+,30648 -- Obsidian Sanctum Fire Cyclone
```

Comment (offl) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql:4]: I think we can open an issue with help-wanted label and ask people for help to add this flag to more creatures, I can add it mostly to TBC without screwing up everything

Comment (jackpoz) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql]: On top of that, we don't have to find all creatures before pushing the sql update. Even if we add 1000 sql updates, as they are applied automatically by worldserver it's all fine

**Pull request merged**